### PR TITLE
Prevent NullPointer on a network with removed IP ranges/"VLANs"

### PR DESCRIPTION
**Pull request by GabrielBrascher**
## Description

When a network IP range is removed, the "vlan" stays mapped on `pod_vlan_map`; therefore, the method that lists the VLANs by pod id will return null VLANS.

This PR adds proper verifications to avoid null pointer exception when deploying VRs on a pod with removed VLANs. The exception was caused on [getPlaceholderNicForRouter](https://github.com/apache/cloudstack/blob/3c2af55d81dbfaf17a3bf0b6247ecd830df12996/server/src/main/java/com/cloud/network/NetworkModelImpl.java#L2306)

```
java.lang.NullPointerException
        at com.cloud.network.NetworkModelImpl.getPlaceholderNicForRouter(NetworkModelImpl.java:2306)
```

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@wido I think if a table has removed column, GenericDao::remove will mark entry as removed otherwise will remove the row.

**Comment by rhtyd** [comment]
@GabrielBrascher can you check the failing travis tests? I've rekicked them twice and kicked them again now.

**Comment by GabrielBrascher** [comment]
@rhtyd @wido It looks like there is an issue on the lines regarding `podVlanMapDao.remove(sc);` that I did not anticipate.

**Comment by rhtyd** [comment]
Can you test/fix it @GabrielBrascher?

**Comment by GabrielBrascher** [comment]
Sure @rhtyd, I will update it soon.

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
larger than normal intermittent failures, will kick again
@blueorangutan package

File: engine/schema/src/main/java/com/cloud/dc/dao/VlanDaoImpl.java [hunk 1]
```diff
@@ -167,7 +167,10 @@ public VlanDaoImpl() {
         List<PodVlanMapVO> vlanMaps = _podVlanMapDao.listPodVlanMapsByPod(podId);
         List<VlanVO> result = new ArrayList<VlanVO>();
         for (PodVlanMapVO pvmvo : vlanMaps) {
-            result.add(findById(pvmvo.getVlanDbId()));
+            VlanVO vlanByPodId = findById(pvmvo.getVlanDbId());
+            if (vlanByPodId != null && vlanByPodId.getRemoved() == null) {
```
Comment (rhtyd) [inline engine/schema/src/main/java/com/cloud/dc/dao/VlanDaoImpl.java]: a null check is enough, removed items are not returned by `GenericDaoBase::findById()`

Comment (GabrielBrascher) [reply engine/schema/src/main/java/com/cloud/dc/dao/VlanDaoImpl.java]: I have added a commit addressing this comment. Thanks for the feedback, @rhtyd!

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Pull request merged**